### PR TITLE
Revert compatibility break with NodePath and ConfigurationTransformation

### PR DESCRIPTION
**Pull request by lucko**
Reverts some breakages introduced in #92

### CODE REVIEW

File: configurate-core/src/main/java/ninja/leaping/configurate/transformation/TransformAction.java [hunk 1]
```diff
@@ -37,6 +37,6 @@
      * @return A modified path, or null if the path is to stay the same
      */
     @Nullable
-    Object[] visitPath(@NonNull NodePath inputPath, @NonNull ConfigurationNode valueAtPath);
+    Object[] visitPath(ConfigurationTransformation.@NonNull NodePath inputPath, @NonNull ConfigurationNode valueAtPath);
```
Comment (dualspiral) [inline configurate-core/src/main/java/ninja/leaping/configurate/transformation/TransformAction.java]: `@NonNull ConfigurationTransformation.NodePath`

Comment (lucko) [reply configurate-core/src/main/java/ninja/leaping/configurate/transformation/TransformAction.java]: Not valid:



The only other way to do it is to add an import statement for 
```java
import ninja.leaping.configurate.transformation.ConfigurationTransformation.NodePath
```

and then have

```java
Object[] visitPath(@NonNull NodePath inputPath, @NonNull ConfigurationNode valueAtPath);
```

but I thought that was unclear, as an interface with the same name already exists in the same package.

Comment (dualspiral) [reply configurate-core/src/main/java/ninja/leaping/configurate/transformation/TransformAction.java]: To comment: the signature of this method will still have changed slightly (it was `SimpleConfigurationTransformation.NodePath`), but it actually points to the same class as `NodePath`'s outer class is `ConfigurationTransformation`, which is the superclass of `SimpleConfigurationTransformation`

I checked the bytecode from Nucleus which was compiled using configurate 3.2, it references `ConfigurationTransformation.NodePath` not `SimpleConfigurationTransformation.NodePath` - so this change should be fine.

**Comment by dualspiral** [comment]
Any stylistic comments asside, I think this looks fine, given the comment I gave before.

@kashike It's probably in our interest to get this wrapped up and pushed out of the door sooner rather than later. I'll let you pull the lever if you're happy with this, as I said in my comment, the bytecode in Nucleus references
`ninja.leaping.configurate.transformation.ConfigurationTransformation$NodePath`, so this should be OK and not an API break.

File: configurate-core/src/main/java/ninja/leaping/configurate/transformation/TransformAction.java (hunk 1 above)
Comment (kashike) [reply configurate-core/src/main/java/ninja/leaping/configurate/transformation/TransformAction.java]: `ConfigurationTransformation. @NonNull NodePath`

**Pull request merged**